blockchain-albatross/src/blockchain.rs

```diff
@@ -1062,12 +1062,12 @@ impl<'env> Blockchain<'env> {
         self.chain_store.get_blocks(start_block_hash, count, include_body, direction, None)
     }
 
-    pub fn get_transactions_root(&self, epoch: u32, txn_option: Option<&Transaction>) -> Option<Blake2bHash> {
-        let mut hashes = Vec::new();
+    pub fn get_epoch_transactions<B, F: Fn(&BlockchainTransaction) -> B>(&self, epoch: u32, f: F, txn_option: Option<&Transaction>) -> Option<Vec<B>> {
```

Comment:
Name type parameter `U` instead of `B`. `B` is used for blocks and is misleading on first glance (`T` is misleading too).

The method name doesn't suggest a function that needs a closure. We can easily return a `impl Iterator` that owns the Transactions. We own the blocks. We just need to match `Block::Micro` and get the transaction `Vec`.

```rust
let iter_first = first_block.transactions().into_iter();
let iter_rest = blocks.into_iter().map(|block| block.transactions()).flatten()
iter_first.chain(iter_rest)
```

Replace `block.transactions()` with something that gets the transactions by ownership. Maybe a method `Block::unwrap_transctions(self) -> Vec<Transaction>` even?
```